### ISSUE TITLE
Failed to add arcticdb for both linux and osx on M1 mac.

### ISSUE THREAD

**Issue by freddong** [prefix-dev/pixi#2185]
### Checks

- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell
pixi init pixi-py --format pyproject
# change pyproject.toml
# from platforms = ["osx-arm64"]
# to platforms = ["linux-64", "osx-arm64"]

pixi add arcticdb
```

pyproject.toml
```toml
[project]
authors = [{name = "XXX", email = "XXXX@gmail.com"}]
dependencies = []
description = "Add a short description here"
name = "pixi-py"
requires-python = ">= 3.11"
version = "0.1.0"

[build-system]
build-backend = "hatchling.build"
requires = ["hatchling"]

[tool.pixi.project]
channels = ["conda-forge"]
platforms = ["linux-64", "osx-arm64"]

[tool.pixi.pypi-dependencies]
pixi_py = { path = ".", editable = true }

[tool.pixi.tasks]

[tool.pixi.dependencies]
arcticdb = ">=4.5.0,<5"

```
### Issue description

pixi 0.30.0, installed with brew.

on M1 mac:
It's OK to install for osx only.
```shell
pixi init pixi-py --format pyproject
pixi add arcticdb
```
The output is Added arcticdb >=4.5.0,<5.

But it failed to add for linux and osx.
```shell
pixi init pixi-py --format pyproject
# change pyproject.toml
# from platforms = ["osx-arm64"]
# to platforms = ["linux-64", "osx-arm64"]

pixi add arcticdb
```

the outpus is:
~/pixi-py $ pixi add arcticdb
⠤ updating lock-file   [00:00:00] [━━━━━━━━━━━━━━━━━━━━━━━─────────────────]    4/7
  ⠂ default:linux-64     [00:00:00] loading repodata                                                                                                                                          thread 'tokio-runtime-worker' panicked at library/core/src/slice/sort/shared/smallsort.rs:862:5:
user-provided comparison function does not correctly implement a total order
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

It crashed.

### Expected behavior

Successful installation for linux and osx.

**Comment by ruben-arts** [comment]
This is because of the `brew` installation, the compile with the latest rust seems to have broken something in the binary. We're not yet building with rust 1.81 so the issue is not fixed. 

Related: https://github.com/prefix-dev/pixi/issues/2122

**Comment by ruben-arts** [comment]
I'm confident this is fixed in `v0.32.1`, so I'm closing it. Please reopen if this was not the case
